src/fortran/dataset.F90

```diff
@@ -87,231 +87,261 @@ module smartredis_dataset
 
 contains
 
-subroutine initialize( this, name )
-  class(dataset_type) :: this
-  character(len=*)    :: name !< Name of the dataset
+subroutine initialize( dataset, name )
```

Comment:
Also should be `self`

Reply:
This is the constructor, so I thought it merited special treatment. Do you disagree?